### PR TITLE
Pins to remote branches

### PR DESCRIPTION
**Pull request by mosteo**
An enhancement on top of #743 and #751. Now you can do

```
[[pins]]
foo = { url="http://blah", branch="bar" }
```

### CODE REVIEW

File: src/alire/alire-user_pins.adb [hunk 1]
```diff
@@ -114,9 +133,10 @@ package body Alire.User_Pins is
             return;
          end if;
 
-         --  Finally update
+         --  Finally update. In case the branch has just been changed by the
+         --  user in the manifest, the following call wil also take care of it.
 
-         if not VCSs.Git.Handler.Update (Destination).Success then
+         if not VCSs.Git.Handler.Update (Destination, Branch).Success then
```
Comment (Fabien-Chouteau) [inline src/alire/alire-user_pins.adb:139]: Will this do a recursive update? (submodules)

File: testsuite/drivers/helpers.py [hunk 2]
```diff
@@ -66,6 +68,14 @@ def dir_separator():
     return '/' if os.name != 'nt' else '\\'
 
 
+def touch(file):
```
Comment (Fabien-Chouteau) [inline testsuite/drivers/helpers.py]: E3 has a touch function: `from e3.os.fs import touch`, it should be available.

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]

File: src/alire/alire-user_pins.adb (hunk 1 above)
Comment (mosteo) [reply src/alire/alire-user_pins.adb:139]: Yes, it's at https://github.com/alire-project/alire/blob/01264f9c9e1785ed19e04d41e9ef5598f2fc5123/src/alire/alire-vcss-git.adb#L436 and https://github.com/alire-project/alire/blob/01264f9c9e1785ed19e04d41e9ef5598f2fc5123/src/alire/alire-vcss-git.adb#L443

File: testsuite/drivers/helpers.py (hunk 2 above)
Comment (mosteo) [reply testsuite/drivers/helpers.py]: OK, I'll use that one then.

Comment (mosteo) [reply testsuite/drivers/helpers.py]: And I'll read through the `e3` APIs, I hadn't realized there were so many things there that can be useful.

**Review by Fabien-Chouteau — APPROVED** [review]

**Pull request merged**